pulse2percept/implants/base.py

```diff
@@ -347,18 +347,20 @@ class ElectrodeGrid(ElectrodeArray):
     * Allow user-specified electrode types (see :issue:`122`).
 
     """
-
-    def __init__(self, shape, x=0, y=0, z=0, rot=0, r=10, spacing=None,
-                 names=('A', '1')):
+    
+    def __init__(self, shape, x=0, y=0, z=0, rot=0, r = 10, spacing=None,
+                 names=('A', '1'), electrode_type = 'DiskElectrode'): # the default electrode type is DiskElectrode
         if not isinstance(names, (tuple, list, np.ndarray)):
             raise TypeError("'names' must be a tuple/list of (rows, cols)")
-
         if not isinstance(shape, (tuple, list, np.ndarray)):
             raise TypeError("'shape' must be a tuple/list of (rows, cols)")
         if len(shape) != 2:
             raise ValueError("'shape' must have two elements: (rows, cols)")
         if np.prod(shape) <= 0:
             raise ValueError("Grid must have all non-zero rows and columns.")
+        if electrode_type != 'DiskElectrode' and electrode_type != "PointSource": # check the value of the electrode_type
```

Comment:
- If type is 'DiskElectrode': make sure `electrode_kwargs` contains an entry for `r` (which is the radius)
- If type is 'PointSource': do nothing
- else: `raise NotImplementedError`